pkg/plugins/runtime/gateway/plugin.go

```diff
@@ -34,6 +38,28 @@ func (p *plugin) BeforeBootstrap(context *core_plugins.MutablePluginContext, con
 	return nil
 }
 
+func (p *plugin) Apply(ctx context.Context, meshContext xds_context.MeshContext, proxy *core_xds.Proxy) error {
+	if proxy.Dataplane == nil || !proxy.Dataplane.Spec.IsBuiltinGateway() {
+		return nil
+	}
+	l, err := gatewayListenerInfoFromProxy(
+		ctx, &meshContext, proxy,
+	)
+	if err != nil {
+		return err
+	}
+	proxy.RuntimeExtensions[metadata.PluginName] = l
+	return nil
+}
+
+func ExtractGatewayListener(proxy *core_xds.Proxy) []GatewayListenerInfo {
```

Comment:
wouldn't it be better to have this function on proxy struct? Then we won't need to import gateway plugin in all policies plugins and you can simply do: `proxy.ExtractGatewayListener()` in policy.

Reply:
That sounds interesting. I thought a little with circular dependency which is why I think I ended up doing this this way but let me double check.

Reply:
Yes so I confirm what I said there's an issue with circular dep also the idea of proxy plugins is that they are type independent so it looks to me that we should keep this kind of method in the proxy plugin